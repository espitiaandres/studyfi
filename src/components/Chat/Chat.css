.outerContainer {
  align-items: center;
  display: flex;
  justify-content: center;
  height: 100vh;
  padding-left: 10px;
  width: 96%;
}
  
.container {
  background: #FFFFFF;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  height: 600px;
  justify-content: space-between;
}

@media (min-width: 320px) and (max-width: 480px) {
  .outerContainer {
    height: 100%;
  }

  .container {
    height: 100%;
    width: 100%;
  }
}

@media (min-width: 480px) and (max-width: 1200px) {
  .container {
    width: 60%;
  }
}
