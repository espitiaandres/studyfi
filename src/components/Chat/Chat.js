//
//  Chat.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React, { useState, useEffect } from 'react';
import queryString from 'query-string';
import io from 'socket.io-client';
import InfoBar from '../InfoBar/InfoBar';
import Input from '../Input/Input';
import Messages from '../Messages/Messages';
import Userslist from '../Userslist/Userslist';
import Join from '../Join/Join';
import './Chat.css';

let socket;

const Chat = ({ location, item }) => {
    const [name, setName] = useState('');
    const [room, setRoom] = useState('');
    const [users, setUsers] = useState([]);
    const [message, setMessage] = useState('');
    const [messages, setMessages] = useState([]);
    const [duplicate, setDuplicate] = useState(false);
    const ENDPOINT = 'https://react-chat-app-back-end.herokuapp.com/';

    useEffect(() => {
        const { name, room } = queryString.parse(location.search);
        socket = io(ENDPOINT, {transports: ['websocket']});
        setName(name);
        setRoom(room);
        socket.emit('join', { name, room }, () => {});

        socket.on('reconnect_attempt', () => {
            socket.io.opts.transports = ['websocket', 'polling'];
        });

        return () => {
            socket.emit('disconnect');
            socket.off();
        };
    }, [ENDPOINT, location.search]);

    useEffect(() => {
        socket.on("roomData", ({ users }) => {
            setUsers(users);
        });

        socket.on('duplicate', (duplicate) => {
            setDuplicate(duplicate.duplicate);
        })
    });

    useEffect(() => {
        socket.on('message', (message) => {
            setMessages(messages => [ ...messages, message ]);
        });
    }, [name]);

    const sendMessage = (event) => {
        event.preventDefault();
        if (message) {
            socket.emit('sendMessage', message, () => setMessage(''));
        }
    }
    return (
        <div>
            {
                duplicate ? 
                <Join duplicate={duplicate}/> :                 
                <div className="outerContainer">
                    <div className="container">
                        <InfoBar room={room}/>
                        <Messages messages={messages} name={name}/>
                        <Input 
                            message={message} setMessage={setMessage} sendMessage={sendMessage} item={item}
                        />
                    </div>
                    <Userslist users={users}/>
                </div>
            }
        </div>
    )
}

export default Chat;
