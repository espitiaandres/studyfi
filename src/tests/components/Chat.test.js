//
//  Chat.test.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import { shallow } from 'enzyme';
import Chat from '../../components/Chat/Chat';

const chatPropsTest = {
    location: "name=testUser&room=testRoom&tz=GMT",
    item: {
        name: "testSong",
        artists: [{
            name: "testArtist"
        }]
    },
    user: "testUser",
    season: true
}

test('should render Chat component correctly', () => {
    const wrapper = shallow(
        <Chat
            location={chatPropsTest.location}
            item={"chatPropsTest.item"}
            user={chatPropsTest.user}
            season={chatPropsTest.season}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
