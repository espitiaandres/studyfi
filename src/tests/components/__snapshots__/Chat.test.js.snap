// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Chat component correctly 1`] = `ShallowWrapper {}`;
